tools/code-analyzer/src/lib/hook-changes.ts

```diff
@@ -47,76 +63,85 @@ export const scanForHookChanges = async (
 	for ( const p in patches ) {
 		const patch = patches[ p ];
 
-		// Separate patches into bits beginning with a comment. If a bit does not have an action, disregard.
-		const patchWithHook = patch.split( '/**' ).find( ( s ) => {
+		// Separate patch into pieces beginning with a comment. If a piece does not have actions/filters discard it.
+		const patchesWithHooks = patch.split( '/**' ).filter( ( s ) => {
 			return s.includes( 'apply_filters' ) || s.includes( 'do_action' );
 		} );
-		if ( ! patchWithHook ) {
-			continue;
-		}
-
-		const results = patchWithHook.match( newRegEx );
 
-		if ( ! results ) {
+		if ( ! patchesWithHooks.length ) {
 			continue;
 		}
 
 		const lines = patch.split( '\n' );
 		const filePath = getFilename( lines[ 0 ] );
 
-		for ( const raw of results ) {
-			// Extract hook name and type.
-			const hookName = raw.match(
-				/(.*)(do_action|apply_filters)\(\s+'(.*)'/
-			);
+		for ( const hookPatch of patchesWithHooks ) {
+			const results = hookPatch.match( newRegEx );
 
-			if ( ! hookName ) {
+			if ( ! results ) {
 				continue;
 			}
 
-			const name = getHookName( hookName[ 3 ] );
-			const description = getHookDescription( raw, name ) || '';
+			for ( const raw of results ) {
+				// Extract hook name and type.
+				const hookName = raw.match(
+					/(.*)(do_action|apply_filters)\(\s+'(.*)'/
+				);
+
+				if ( ! hookName ) {
+					continue;
+				}
 
-			const hookType =
-				hookName[ 2 ] === 'do_action' ? 'action' : 'filter';
+				const name = getHookName( hookName[ 3 ] );
+				const description = getHookDescription( raw, name ) || '';
 
-			const changeType = getHookChangeType( raw );
+				const hookType =
+					hookName[ 2 ] === 'do_action' ? 'action' : 'filter';
 
-			if ( ! hookName[ 2 ].startsWith( '-' ) ) {
-				let ghLink = '';
+				const changeType = getHookChangeType( raw );
 
-				try {
-					const data = await fs.promises.readFile(
-						tmpRepoPath + filePath,
-						'utf-8'
-					);
+				if ( ! hookName[ 2 ].startsWith( '-' ) ) {
+					let ghLink = '';
 
-					const reg = new RegExp( name );
+					try {
+						// If path doesn't exist for some reason don't try create a link.
+						if ( existsSync( tmpRepoPath + filePath ) ) {
```

Comment:
This helps me for testing, I need to avoid trying to read files that don't exist.

My justification for the change is that, if we couldn't read a file for some reason to generate a github link there is not a reason to crash anyway. Open to debate :D